lib/internal/modules/esm/load.js

```diff
@@ -141,7 +141,9 @@ async function defaultLoad(url, context = kEmptyObject) {
   }
 
   validateAttributes(url, format, importAttributes);
-
+  if (format == null && getOptionValue('--experimental-detect-module')) {
```

Comment:
And what if we detected that the module is ESM? Then `format` would be `'module'` already?

Reply:
Yes

Reply:
Essentially, if we don't know the format and we should autodetect it, assume it is commonjs